manifests/vhost/custom.pp

```diff
@@ -0,0 +1,38 @@
+# See README.md for usage information
+define apache::vhost::custom(
+  $content,
+  $ensure = 'present',
+  $priority = '25',
+) {
+  include ::apache
+
+  ## Apache include does not always work with spaces in the filename
+  $filename = regsubst($name, ' ', '_', 'G')
+
+  ::apache::custom_config { $filename:
+    ensure   => $ensure,
+    confdir  => $::apache::vhost_dir,
+    content  => $content,
+    priority => $priority,
+  }
+
+  # NOTE(pabelanger): This code is duplicated in ::apache::vhost and needs to
+  # converted into something generic.
+  if $::apache::vhost_enable_dir {
+    $vhost_symlink_ensure = $ensure ? {
+      present => link,
+      default => $ensure,
+    }
+
+    file { "${priority}-${filename}.conf symlink":
+      ensure  => $vhost_symlink_ensure,
+      path    => "${::apache::vhost_enable_dir}/${priority}-${filename}.conf",
+      target  => "${::apache::vhost_dir}/${priority}-${filename}.conf",
+      owner   => 'root',
+      group   => $::apache::params::root_group,
+      mode    => '0644',
+      require => Apache::Custom_config[$filename],
+      notify  => Class['apache::service'],
```

Comment:
So.. I believe in the `apache::custom_config` class, there is a conditional that will set the notify if `$verify_config` is `true`. Do we want to adhere to that here?

Also corollary to that... since this is merely a wrapper around `apache::custom_config` should the notify be omitted here? So that you don't a) trigger an unwanted refresh or b) trigger a secondary refresh. Could there could be some idempotency issues with that as well?